internal/cli/atlas/streams/connection/describe.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code was autogenerated at 2023-07-04T05:39:19Z. Note: Manual updates are allowed, but may be overwritten.
+
+package connection
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type DescribeOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	id              string
+	streamsInstance string
+	store           store.StreamsConnectionDescriber
+}
+
+func (opts *DescribeOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+var describeTemplate = `NAME	TYPE	INSTANCE	SERVERS
+{{.Name}}	{{.Type}}	{{.Instance}}	{{.Servers}}
+`
+
+func (opts *DescribeOpts) Run() error {
+	r, err := opts.store.StreamConnection(opts.ConfigProjectID(), opts.streamsInstance, opts.id)
+	if err != nil {
+		return err
+	}
+
+	if opts.IsJSONOutput() || opts.IsJSONPathOutput() {
+		return opts.Print(r.StreamsConnection)
+	}
+
+	return opts.Print(r)
```

Comment:
am I bit worry about this in terms of support, in particular having to explain the conditions when the output is different depending on flags is hard for users to grasp, also there's the question what do we do with go template? what struct would we document?

Reply:
Yeah, as discussed this is not ideal (was originally proposed by the PMs, but this isn't idomatic to how atlas-cli works). 

Resolved in 2dd56ad6, I'm now just using go templates to select the right field, and ignoring instance entirely.